fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerIndexFromIndexTest.java

```diff
@@ -1110,4 +1112,94 @@ public void testIndexFromIndexRebuild() {
             context.commit();
         }
     }
+
+    @Test
+    public void testIndexFromIndexBlock() {
+        // start indexing by Index, verify continuation
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        final int numRecords = 49;
+        final int chunkSize  = 12;
+        final int numChunks  = 1 + (numRecords / chunkSize);
+
+        Index srcIndex = new Index("src_index", field("num_value_2"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS);
+        Index tgtIndex = new Index("tgt_index", field("num_value_3_indexed"), IndexTypes.VALUE);
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = myHook(srcIndex, tgtIndex);
+
+        openSimpleMetaData();
+        populateData(numRecords);
+
+        openSimpleMetaData(hook);
+        buildSrcIndex(srcIndex);
+
+        // Partly build index
+        openSimpleMetaData(hook);
+        buildIndexAndCrashHalfway(tgtIndex, chunkSize, 1, timer,
+                OnlineIndexer.IndexingPolicy.newBuilder()
+                        .setSourceIndex("src_index")
+                        .forbidRecordScan()
+                        .build());
+
+        openSimpleMetaData(hook);
+        String luka = "my name is Luka";
+        try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setSubspace(subspace)
+                .setIndex(tgtIndex)
+                .build()) {
+            indexer.indexingStamp(OnlineIndexer.IndexingStampOperation.BLOCK, luka, 10);
+        }
+
+        // query, ensure blocked
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setSubspace(subspace)
+                .setIndex(tgtIndex)
+                .build()) {
+            final AbstractMap<String, IndexBuildProto.IndexBuildIndexingStamp> stampMap =
+                    indexer.indexingStamp(OnlineIndexer.IndexingStampOperation.QUERY, null, 0);
+            System.out.println(stampMap);
+            String indexName = tgtIndex.getName();
+            final IndexBuildProto.IndexBuildIndexingStamp stamp = stampMap.get(indexName);
+            assertEquals(IndexBuildProto.IndexBuildIndexingStamp.Method.BY_INDEX, stamp.getMethod());
+            assertTrue(stamp.getBlock());
+            assertEquals(luka, stamp.getBlockID());
+            assertTrue(stamp.getBlockExpireEpochSeconds() > (System.currentTimeMillis() / 1000));
+            assertTrue(stamp.getBlockExpireEpochSeconds() < 20 + (System.currentTimeMillis() / 1000));
```

Comment:
You use `/ 1000` in a lot of places to convert from to/from millis. It would be better to use `TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis())` (or it might be better to just have the stamp have millis since epoch, as I commented elsewhere, and avoid all this conversion)

Reply:
✅